### PR TITLE
Add description of Azure spot VMs for kubectl-gs template nodepool

### PR DESCRIPTION
**Pull request by whites11**
Towards: https://github.com/giantswarm/roadmap/issues/372

This PR adds 2 more flags to the kubectl-gs template nodepool docs (azure spot VMs)

### Please remember to

- [x] Give the PR a meaningful title -- it will be shown in the release notes
- [ ] Add (or remove) [user questions](https://github.com/giantswarm/docs/blob/master/CONTRIBUTING.md#front-matter) associated with any updated docs

### CODE REVIEW

File: src/content/ui-api/kubectl-gs/template-nodepool.md [hunk 1]
```diff
@@ -62,6 +62,8 @@ Here are the supported flags:
 ### Azure specific
 
 - `--azure-vm-size` - Azure VM size to use for workers (e.g. *Standard_D4s_v3*).
+- `--azure-spot-vms` - Wheter to use Spot VMs for this Node Pool or not (defaults to false that means not to use Spot VMs).
```
Comment (marians) [inline src/content/ui-api/kubectl-gs/template-nodepool.md:65]:
```suggestion
- `--azure-spot-vms` - Whether to use spot VMs for this node pool (defaults to false which means not to use spot VMs).
```

File: src/content/ui-api/kubectl-gs/template-nodepool.md [hunk 2]
```diff
@@ -62,6 +62,8 @@ Here are the supported flags:
...
+- `azure-spot-max-price` - Max hourly price in USD to pay for one Spot VM. If the current price for the VM is exceeded, the VM is deleted.
```
Comment (marians) [inline src/content/ui-api/kubectl-gs/template-nodepool.md:66]:
```suggestion
- `azure-spot-max-price` - Max hourly price in USD to pay for one spot VM. If the current price for the VM is exceeded, the VM is deleted.
```

Comment (marians) [reply src/content/ui-api/kubectl-gs/template-nodepool.md]: I'd like us to document what the default is. I. e. what price is used if this flag is not used.

File: src/content/ui-api/kubectl-gs/template-nodepool.md [hunk 3]
```diff
@@ -62,6 +62,8 @@ Here are the supported flags:
 ### Azure specific
 
 - `--azure-vm-size` - Azure VM size to use for workers (e.g. *Standard_D4s_v3*).
+- `--azure-spot-vms` - Whether to use spot VMs for this node pool (defaults to false which means not to use spot VMs).
+- `azure-spot-max-price` - Max hourly price in USD to pay for one spot VM. If the current price for the VM is exceeded, the VM is deleted.
```
Comment (axbarsan) [inline src/content/ui-api/kubectl-gs/template-nodepool.md:66]:
```suggestion
- `--azure-spot-max-price` - Max hourly price in USD to pay for one spot VM. If the current price for the VM is exceeded, the VM is deleted.
```

**Comment by marians** [comment]
Did you miss my comment

> I'd like us to document what the default is. I. e. what price is used if this flag is not used.

regarding `--azure-spot-max-price` ?

**Comment by whites11** [comment]
> Did you miss my comment
> 
> > I'd like us to document what the default is. I. e. what price is used if this flag is not used.
> 
> regarding `--azure-spot-max-price` ?

yes, looking at them now

**Review by axbarsan — APPROVED** [review]

**Comment by whites11** [comment]
> > Did you miss my comment
> > > I'd like us to document what the default is. I. e. what price is used if this flag is not used.
> > 
> > 
> > regarding `--azure-spot-max-price` ?
> 
> yes, looking at them now

done

File: src/content/ui-api/kubectl-gs/template-nodepool.md [hunk 4]
```diff
@@ -62,6 +62,8 @@ Here are the supported flags:
 ### Azure specific
 
 - `--azure-vm-size` - Azure VM size to use for workers (e.g. *Standard_D4s_v3*).
+- `--azure-spot-vms` - Whether to use spot VMs for this node pool (defaults to false which means not to use spot VMs).
+- `--azure-spot-max-price` - Max hourly price in USD to pay for one spot VM. If the current price for the VM is exceeded, the VM is deleted. If not set, the on demand price for the same instance size is used as max value.
```
Comment (marians) [inline src/content/ui-api/kubectl-gs/template-nodepool.md:66]:
```suggestion
- `--azure-spot-max-price` - Max hourly price in USD to pay for one spot VM. If the current price for the VM is exceeded, the VM is deleted. If not set, the on-demand price for the same machine size is used as limit.
```

**Review by marians — APPROVED** [review]

**Comment by marians** [comment]
Normally we should wait with publishing until the related product change is merged and released.

**Pull request merged**